### DISCUSSION TITLE
How can you detect the size/position of an image inside a canvas?

### DISCUSSION THREAD

**Discussion by r2997790** [ImageMagick/ImageMagick#5345] [Digital Image Processing] [answered]
Hi everyone,

Does anyone know how to detect the size of an image inside a canvas? For example, image you have 400x400px canvas, and you have a 300px square image inside it.... Is there a way for ImageMagick to match against a sample image and say that it appears in the image, and it is X size and where is it positioned on the canvas?

I'm asking because I need to replace a bunch of images in a collection of files and I want to overwrite them to the same size and dimension files. 

![pos](https://user-images.githubusercontent.com/29481754/180191042-b74da477-780d-4bff-a27d-42c42cbae973.png)
 
Thank you for any advice on this!

R

**Comment by snibgo** [comment, marked as answer]
We can trim the plain border. Using your image:
```
magick imgCanv.png -trim x.png

magick imgCanv.png -trim info:
imgCanv.png PNG 308x310 400x400+42+40 8-bit sRGB 6238B 0.000u 0:00.000

magick imgCanv.png -format %@ info:
308x310+42+40
f:\web\im>
```
The trimmed image is 308x310 pixels, and starts at (42,40) in the untrimmed image.

**Reply by r2997790** [reply]
Brilliant. Thank you very much.
